resources/metadata/gitopsUpdateDeployment.yaml

```diff
@@ -0,0 +1,112 @@
+metadata:
+  name: gitopsUpdateDeployment
+  description: Updates Kubernetes Deployment Manifest in Infrastructure Git Repo
+  longDescription: |
+    This step allows you to update the deployment manifest in another repository.
+
+    It can for example be used for GitOps scenarios where the update of the manifests triggers an update of the corresponding deployment in Kubernetes.
+spec:
+  inputs:
+    secrets:
+      - name: gitHttpsCredentialsId
+        description: Jenkins 'Username with password' credentials ID containing username/password for http access to your git repository.
+        type: jenkins
+    params:
+      - name: branchName
+        description: Currently ignored and only the main branch is considered. The name of the branch you want the changes pulled into.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        type: string
+        mandatory: true
```

Comment:
Lets comments this till implementation is there.

```suggestion
#      - name: branchName
#        description: Currently ignored and only the main branch is considered. The name of the branch you want the changes pulled into.
#        scope:
#          - PARAMETERS
#          - STAGES
#          - STEPS
#        type: string
#        mandatory: true
```

Reply:
I have implemented it Today morning